### PR TITLE
Feature: reverse sorting

### LINKED ISSUE #254: Allow reverse sorting
Any sort action should work in ascending and descending order.

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #254 

### Give a longer description of what this PR addresses and why it's needed
This PR adds reverse sorting rows for size and degree.

To do so, an optional field `sortByOrder` is added to relevant sorting functions in `packages/core`. 

Additionally, a new field is added to the UpSet Grammar: sortByOrder (default `Ascending`). This field only applies to row sorting, is directly beneath `sortBy` and can be either `Ascending` or `Descending`. The addition of this field should not affect existing use cases of the UpSet Grammar, as existing fields are unchanged.

No action is added in Trrack, rather, an optional field is added to the `sortBy` action, where the config field is set.
The provenance tree action is some variation of "Sort by Size, Ascending"

### Provide pictures/videos of the behavior before and after these changes (optional)
Provenance tree example:
image

Behavior example:
upset-reverse-sort

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [X] test edge cases

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -86,7 +86,9 @@ export const aggregateByList = [
 export type AggregateBy = typeof aggregateByList[number];
 
 export const sortByList = ['Degree', 'Size', 'Deviation'] as const;
+export const sortByOrderList = ['Ascending', 'Descending'] as const;
 export type SortBy = typeof sortByList[number];
+export type SortByOrder = typeof sortByOrderList[number];
```
Comment (JackWilb) [inline packages/core/src/types.ts:91]: Could this be simplified?
```suggestion
export type SortBy = typeof sortByList[number];
export type SortByOrder = 'Ascending' | 'Descending';
```

File: packages/upset/src/components/Sidebar.tsx [hunk 2]
```diff
@@ -97,33 +106,88 @@ export const Sidebar = () => {
...
-                    css={itemDivCSS}
-                    key={sort}
-                    aria-label={helpText.sorting[sort]}
-                    onKeyDown={(e) => {
-                      if (e.key === 'Enter') {
-                        actions.sortBy(sort);
+              <Box
+                css={itemDivCSS}
+                key="Degree"
+                aria-label={helpText.sorting['Degree']}
```
Comment (JackWilb) [inline packages/upset/src/components/Sidebar.tsx:115]: Does this help text need to be updated for ascending and descending?

File: packages/upset/src/components/Sidebar.tsx [hunk 3]
```diff
@@ -97,33 +106,88 @@ export const Sidebar = () => {
...
-                    <FormControlLabel
-                      key={sort}
-                      value={sort}
-                      label={sort}
-                      control={<Radio size="small" />}
-                    />
-                    <HelpCircle text={helpText.sorting[sort]} />
-                  </Box>
-                )))}
+                    </Box>
```
Comment (JackWilb) [inline packages/upset/src/components/Sidebar.tsx:145]: How is this for accessibility? Would a blind person be able to tell the sort direction from this html?

Comment (JackWilb) [reply packages/upset/src/components/Sidebar.tsx:145]: Same for the size sort

File: packages/upset/src/components/Sidebar.tsx [hunk 4]
```diff
@@ -97,33 +106,88 @@ export const Sidebar = () => {
...
+                          }
+                        }
+                      }}
+                    >
+                      <Typography>Size</Typography>
+                      { sortBy === 'Size' &&
+                        (sortByOrder === 'Ascending' ?
+                          <ArrowUpward sx={arrowIconCSS} /> :
+                          <ArrowUpward sx={{ ...arrowIconCSS, transform: 'rotate(180deg)' }} />
+                        )
```
Comment (JackWilb) [inline packages/upset/src/components/Sidebar.tsx:182]: Let's move the ternary to the CSS?
```suggestion
                        <ArrowUpward sx={{ ...arrowIconCSS, transform: (sortByOrder === 'Ascending' ? '' : 'rotate(180deg)') }} />
```

Comment (JackWilb) [reply packages/upset/src/components/Sidebar.tsx:182]: We could do this for both if you think it's cleaner/better for react rendering

**Review by JackWilb — COMMENTED** [review]
Some notes, and waiting on your changes

**Review by JackWilb — APPROVED** [review]
